core/vdbe/insn.rs

```diff
@@ -488,3 +488,164 @@ pub enum Insn {
         where_clause: String,
     },
 }
+
+impl Insn {
+    // returns a function that evaluates a binary expression of final values
+    // do not use on unresolved AggFunc values not in their .final_state()
+    pub fn math_binary_eval(&self) -> fn(&OwnedValue, &OwnedValue) -> OwnedValue {
```

Comment:
this doesn't need to return a function, right?

```
pub fn math_binary_eval(&self, lhs: &mut OwnedValue, rhs: &mut OwnedValue) -> OwnedValue {
  // ... coerce Agg to primitive here ..
  match self {
     ...
  }
}
```

Reply:
yup most definitely :+1: , got caught in x-y problem. was able to rm another 200  lines or so